fixtures/ext-types/guid/src/lib.rs

```diff
@@ -1,13 +1,51 @@
 // A trivial guid.
 pub struct Guid(pub String);
 
+// This error is represented in the UDL.
+#[derive(Debug, thiserror::Error)]
+pub enum GuidError {
+    #[error("The Guid is too short")]
+    TooShort,
+}
+
+// This error is not represented in the UDL - it's only to be used internally (although
+// for test purposes, we do allow this to leak out below.)
+#[derive(Debug, thiserror::Error)]
+pub enum InternalError {
+    #[error("Something unexpected went wrong")]
+    Unexpected,
+}
+
 fn get_guid(guid: Option<Guid>) -> Guid {
+    // This function doesn't return a Result, so all conversion errors are panics
     match guid {
-        Some(guid) => guid,
+        Some(guid) => {
+            assert!(
+                !guid.0.is_empty(),
+                "our UniffiCustomTypeWrapper already checked!"
+            );
```

Comment:
IIUC if this assertion fails, then that would be a test error.  I think this string would make it hard to figure out what went wrong in that case though.  How about changing the it to "our UniffiCustomTypeWrapper failed to check for an empty GUID" or something like that.